server/src/main/java/io/druid/server/security/AuthorizationUtils.java

```diff
@@ -0,0 +1,337 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.security;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import io.druid.java.util.common.ISE;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Static utility functions for performing authorization checks.
+ */
+public class AuthorizationUtils
+{
+  public final static Access ACCESS_OK = new Access(true, "All resource-actions authorized.");
+
+  /**
+   * Check a resource-action using the authorization fields from the request.
+   *
+   * Otherwise, if the resource-actions is authorized, return ACCESS_OK.
+   *
+   * This function will set the DRUID_AUTH_TOKEN_CHECKED attribute in the request.
+   *
+   * If this attribute is already set when this function is called, an exception is thrown.
+   *
+   * @param request HTTP request to be authorized
+   * @param resourceAction A resource identifier and the action to be taken the resource.
+   * @param authorizerMapper The singleton AuthorizerMapper instance
+   * @return ACCESS_OK or the failed Access object returned by the Authorizer that checked the request.
+   */
+  public static Access authorizeResourceAction(
+      final HttpServletRequest request,
+      final ResourceAction resourceAction,
+      final AuthorizerMapper authorizerMapper
+  )
+  {
+    return authorizeAllResourceActions(
+        request,
+        Lists.newArrayList(resourceAction),
+        authorizerMapper
+    );
+  }
+
+  /**
+   * Check a list of resource-actions using the authorization fields from the request.
+   *
+   * If one of the resource-actions fails the authorization check, this method returns the failed
+   * Access object from the check.
+   *
+   * Otherwise, return ACCESS_OK if all resource-actions were successfully authorized.
+   *
+   * This function will set the DRUID_AUTH_TOKEN_CHECKED attribute in the request.
+   *
+   * If this attribute is already set when this function is called, an exception is thrown.
+   * @param request HTTP request to be authorized
+   * @param resourceActions A list of resource-actions to authorize
+   * @return ACCESS_OK or the Access object from the first failed check
+   */
+  public static Access authorizeAllResourceActions(
+      final HttpServletRequest request,
+      final List<ResourceAction> resourceActions,
+      final AuthorizerMapper authorizerMapper
+  )
+  {
+    final String identity = (String) request.getAttribute(AuthConfig.DRUID_AUTH_TOKEN);
+    if (identity == null) {
+      throw new ISE("Null identity.");
+    }
+
+    final String namespace = (String) request.getAttribute(AuthConfig.DRUID_AUTH_NAMESPACE);
```

Comment:
Hm, I think I prefer the configuration form I described for the following reasons:
- With druid.auth.authenticators containing type names, there can only be one instance of each Authenticator type
- With druid.auth.authorizerMap, each Authenticator is attached to a single Authorizer, I think it's more flexible to let the Authenticator implementation choose what authorizer name(s) to route requests to
- I'm not a big fan of using a JSON map in the runtime properties for readability reasons, I feel like that configuration format is geared towards single line properties